crates/primitives/src/transaction/signature.rs

```diff
@@ -67,6 +60,27 @@ impl Signature {
         }
     }
 
+    /// Output the length of the signature without the length of the RLP header
+    pub(crate) fn payload_len(&self) -> usize {
+        self.odd_y_parity.length() + self.r.length() + self.s.length()
+    }
+
+    /// Encode the `odd_y_parity`, `r`, `s` values without a RLP header.
+    pub(crate) fn encode(&self, out: &mut dyn reth_rlp::BufMut) {
+        self.odd_y_parity.encode(out);
+        self.r.encode(out);
+        self.s.encode(out);
+    }
+
+    /// Decodes the `odd_y_parity`, `r`, `s` values without a RLP header.
+    pub(crate) fn decode(buf: &mut &[u8]) -> Result<Self, DecodeError> {
```

Comment:
because the new methods `encode_legacy`, `decode_legacy`, and `payload_len_legacy` now support both EIP-155 and legacy signatures, I don't think we need these methods any more - we can just use the others, because they are compatible with both formats

Reply:
These functions are used for Non Legacy transactions, EIP2930 and EIP1559.